buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -354,7 +346,7 @@ private static void autoConfigureClusterDependencies(
             })));
     }
 
-    private static void configureCleanupHooks(Project project) {
+    private void configureCleanupHooks(Project project) {
```

Comment:
We can ditch the `synchronized` block here since this is being done at configuration time.